### PR TITLE
extend coinjoin logs

### LINKED ISSUE #7376: Report reason for failing to sign coinjoin
Suite should report to SatoshiLabs any failures which cause a coinjoin round to break. A round is broken if one or more of the clients fail to provide their transaction signatures to the coinjoin coordinator. Possible reasons that we need to watch out for and report are:
- Internet disconnection during the critical phase.
- Tor failure during the critical phase.
- Forced Suite shutdown during the critical phase.
- Coordinator did not provide `affiliateRequest`.
- Trezor disconnection during the critical phase.
- Trezor error during signing (including the error message), e.g. `Invalid signature in CoinJoin request` or `Invalid external input`.
- Trezor takes too long to sign, i.e. Suite cannot return the transaction signature within the deadline.
- Big latency in Tor. (Not sure if we have a way to detect this on the client side, but let's try.)
- Other?

These events should be reported to us via Sentry. Reporting via the analytics system seems unsuitable, because it allows opting out. We don't want to allow opting out in this case, because diagnosing the round breakage is too important. Nevertheless, we should do maximum for user privacy by reporting these events via Tor (except for actual Tor failures, which will have to go through clearnet).

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

- [feat(coinjoin): extend CoinjoinBackend logging](https://github.com/trezor/trezor-suite/commit/01b093d34560874d94b1399213d97699d7f33ba0)
  logger extension cherrypicked from https://github.com/trezor/trezor-suite/pull/7571 as inpiration
  CoinjoinBackend emits `log` event  as object `{ level: 'log' | 'warn' | 'error' | 'debug', payload: string }`
- [feat(coinjoin): extend CoinjoinClient logging](https://github.com/trezor/trezor-suite/commit/752c0afe438c44135f094207b54a057120d735cc)
  Doing the same for `CoinjoinClient` with additional redaction of sensitive messages
- [feat(coinjoin): add CoinjoinClient.getRoundsInCriticalPhase method](https://github.com/trezor/trezor-suite/commit/ca9d88c284539b81e18407ba24c160deb1c52c5c)
  add new method to CoinjoinClient, used in following commit (emit CoinjoinClient exceptions)
- [chore(suite): handle extended CoinjoinClient events](https://github.com/trezor/trezor-suite/commit/d3ab012aab2faf8d38c3e996bb4efaa2e3758e50)
  self explanatory
- [chore(suite): emit CoinjoinClient exceptions](https://github.com/trezor/trezor-suite/commit/ad51abfce12c956efcbb99b11ab596819faa3a59)
  emit certain CoinjoinClient errors from suite renderer
 - tests of coinjoinClientActions
   create and use shared mock of `coinjoinService`
   test exceptions emitted from actions and coinjoinMiddleware
 
 - [x] log errors to sentry
 All coinjoin errors can be found in Sentry using [`coinjoinReport:True ` filter](https://satoshilabs.sentry.io/issues/?query=is%3Aunresolved+coinjoinReport%3ATrue&referrer=issue-list&statsPeriod=14d). 
 Example: https://satoshilabs.sentry.io/share/issue/98a40a38d0f94191b1eaa64e337ec28d/

## List of exceptions:

- [x] [suite renderer coinjoinMiddleware] Internet disconnection during the critical phase.
- [x] [suite renderer coinjoinMiddleware] Tor failure during the critical phase.
- [x] [suite desktop main] Forced Suite shutdown during the critical phase.
- [x] [suite renderer action] Trezor disconnection during the critical phase.
- [x] [coinjoin transactionSigning] Trezor error during signing (including the error message), e.g. Invalid signature in CoinJoin request or Invalid external input.
- [x] [coinjoin transactionSigning] Trezor takes too long to sign, i.e. Suite cannot return the transaction signature within the deadline.
> It's hard to tell if signature was not sent in time because of Trezor signing process or network failure or any other reason.
> we will just log signTransaction time for each failed round

- [x] [request-manager] Big latency in Tor. (Not sure if we have a way to detect this on the client side, but let's try.)
> timeouted requests will be logged automatically in each phase

- [x] [coinjoin outputRegistration] Change address already used, coordinator rejects output address with AlreadyRegisteredScript
- [x] [coinjoin outputRegistration] Not enough change addresses available, extending change addresses derived gap, result of error above
- [x] [coinjoin transactionSigning] Coordinator did not provide affiliateRequest.
  

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/7376

### CODE REVIEW

File: packages/suite-desktop/src/modules/coinjoin.ts [hunk 1]
```diff
@@ -54,7 +54,13 @@ export const init: Module = ({ mainWindow }) => {
             // override default url in coinjoin settings
             settings.middlewareUrl = coinjoinMiddleware.getUrl();
             const client = new CoinjoinClient(settings);
-            client.on('log', message => logger.debug(SERVICE_NAME, message));
+            client.on('log', ({ level, payload }) => {
+                if (level === 'log') {
```
Comment (szymonlesisz) [inline packages/suite-desktop/src/modules/coinjoin.ts:67]: ?

File: packages/coinjoin/src/types/logger.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export type LogLevel = 'debug' | 'log' | 'warn' | 'error';
+
+export type LogEvent = {
+    level: LogLevel;
+    payload: unknown[];
+};
+
+export interface Logger {
+    debug(message: unknown, ...params: unknown[]): void;
```
Comment (szymonlesisz) [inline packages/coinjoin/src/types/logger.ts]: string

File: packages/coinjoin/src/client/round/connectionConfirmation.ts [hunk 3]
```diff
@@ -142,7 +143,7 @@ export const confirmationInterval = (
                 ) {
                     input.setError(error);
                 }
-                options.log(`Confirmation interval with error ${error.message}`);
+                logger.log(`Confirmation interval with error ${error.message}`);
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/connectionConfirmation.ts]: warn

File: packages/coinjoin/src/client/round/inputRegistration.ts [hunk 4]
```diff
@@ -25,18 +25,19 @@ const registerInput = async (
     prison: CoinjoinPrison,
     options: CoinjoinRoundOptions,
 ): Promise<Alice> => {
+    const { logger } = options;
     if (input.error) {
-        options.log(`Trying to register input with error ${input.error}`);
+        logger.log(`Trying to register input with error ${input.error}`);
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/inputRegistration.ts]: warn

File: packages/coinjoin/src/client/round/inputRegistration.ts [hunk 5]
```diff
@@ -77,7 +78,7 @@ const registerInput = async (
             },
         )
         .catch(error => {
-            options.log(
+            logger.log(
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/inputRegistration.ts]: warn

File: packages/coinjoin/src/client/round/outputRegistration.ts [hunk 6]
```diff
@@ -73,6 +74,7 @@ const registerOutput = async (
                 if (
                     error.message === coordinator.WabiSabiProtocolErrorCode.AlreadyRegisteredScript
                 ) {
+                    logger.error(`Change address already used ${round.id}`);
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/outputRegistration.ts]: redact round.id?

File: packages/coinjoin/src/client/round/selectRound.ts [hunk 7]
```diff
@@ -301,7 +303,7 @@ export const selectInputsForRound = async ({
                         )
                         .then(indices => indices.filter(i => utxos[i])) // filter valid existing indices
                         .catch(error => {
-                            options.log(`selectInputsForRound failed ${error.message}`);
+                            logger.log(`selectInputsForRound failed ${error.message}`);
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/selectRound.ts]: error

File: packages/coinjoin/src/client/round/selectRound.ts [hunk 8]
```diff
@@ -349,14 +351,14 @@ export const selectRound = async ({
...
+    const { logger, setSessionPhase } = options;
 
     const unregisteredAccounts = getUnregisteredAccounts({ accounts, coinjoinRounds, options });
     const unregisteredAccountKeys = unregisteredAccounts.map(({ accountKey }) => accountKey);
 
-    log('Looking for rounds');
+    logger.log('Looking for rounds');
     if (!runningAffiliateServer) {
-        log('Affiliate server is not running. Round selection ignored');
+        logger.log('Affiliate server is not running. Round selection ignored');
```
Comment (szymonlesisz) [inline packages/coinjoin/src/client/round/selectRound.ts]: error? or warn?

File: packages/suite/src/actions/wallet/coinjoinAccountActions.ts [hunk 9]
```diff
@@ -525,21 +532,28 @@ export const pauseCoinjoinSession =
...
+        const state = getState();
         const {
             devices,
-            wallet: { accounts, coinjoin },
-        } = getState();
+            wallet: { accounts },
+        } = state;
 
-        const disconnectedDevices = devices.filter(d => d.id === deviceID && d.remember);
+        const disconnectedDevices = devices.filter(d => d.id === deviceID && d.remember); // TODO: remeber?
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/coinjoinAccountActions.ts]: resolve this todo?

**Review by matejkriz — APPROVED** [review]

**Pull request merged**